### PR TITLE
Adding a wrapper to the lean backend to be able to handle more functions

### PR DESCRIPTION
**Pull request by lfrenot**
This adds a wrapper to the lean backend, which allows handling additional bitvector operations.
The implemented operations are: length, signExtend, zeroExtend, truncate and truncateLSB.

The wrapper functions are in the namespace Sail.BitVec to avoid collisions.

The remaining unimplemented operations require translating type conditions into Lean, which is not handled for now.

### CODE REVIEW

**Comment by lfrenot** [comment]
This fails because I have to updated the tests to match the new output.
To continue I'd like to have some pointers on:
- Where to put the wrapper file(s)
- How to edit [sail_plugin_lean.ml](https://github.com/rems-project/sail/pull/801/files#diff-30cea0166ca43e4170eb074fb7638ebe6bdbb7054466bc7152a496056b5a41df) to add those files to the generated output

File: test/lean/bitvec_operation.sail [hunk 1]
```diff
@@ -47,3 +72,12 @@ function bitvector_xor(x, y) = {
...
 
+val bitvector_unsigned : bits(16) -> nat
+function bitvector_unsigned(x) = {
+  unsigned (x)
+}
+
+val bitvector_signed : bits(16) -> int
+function bitvector_signed(x) = {
+  signed (x)
+}
```
Comment (tobiasgrosser) [inline test/lean/bitvec_operation.sail]: The line terminator seems to be wrong here?

Comment (lfrenot) [reply test/lean/bitvec_operation.sail]: Yes sorry, fixed

**Review by tobiasgrosser — APPROVED** [review]
This looks good from my side. Let's see what @bacam and @Alasdair say.

File: src/sail_lean_backend/pretty_print_lean.ml [hunk 2]
```diff
@@ -76,6 +76,7 @@ let rec doc_typ (Typ_aux (t, _) as typ) =
   | Typ_id (Id_aux (Id "int", _)) -> string "Int"
   | Typ_id (Id_aux (Id "bool", _)) -> string "Bool"
   | Typ_id (Id_aux (Id "bit", _)) -> parens (string "BitVec 1")
+  | Typ_id (Id_aux (Id "nat", _)) -> string "Nat"
```
Comment (bacam) [inline src/sail_lean_backend/pretty_print_lean.ml:79]: Can you use a `Nat` type where an `Int` is expected in Lean?  Sail treats nats as a form of integer.

**Review by bacam — APPROVED** [review]

File: src/sail_lean_backend/pretty_print_lean.ml (hunk 2 above)
Comment (lfrenot) [reply src/sail_lean_backend/pretty_print_lean.ml:79]: Yes, `Nat` are coerced automatically to `Int` by Lean where needed

**Pull request merged**